base/components/Toc.js

```diff
@@ -27,32 +27,49 @@ import { trees } from 'routes'
 
 const getRootPath = pathname => `/${pathname.split('/')[1]}`
 
-const renderRoute = (route, i, pathname) => (
-  <div key={i}>
+const SafeLink = ({className, name, path, pathname}) => {
```

Comment:
I'm not sure I understand what SafeLink is. It seems like it replaces the link with text if the current page is the destination?

Reply:
Agreed, a short comment before each component explaining what its purpose is could be helpful to casual readers.

Reply:
thanks. I'm going to add a comment in the code. 
The idea is that if you click on a <Link> that has a to prop equivalent to the current path, the "history" module is going to generate warnings. To avoid that I'm using the SafeLink wrapper which only creates a <Link> component for something that would take the user to a different route.